torch/csrc/jit/codegen/cuda/scheduler/utils.cpp

```diff
@@ -1134,6 +1096,40 @@ IterDomain* projectIdToRoot(
 }
 } // namespace
 
+IterDomain* innerMostRootDim(TensorView* tv) {
+  const auto& root_domain = tv->hasReduction() && tv->hasRFactor()
+      ? tv->getRootDomain()
+      : tv->getMaybeRFactorDomain();
```

Comment:
I think this is reversed to how we normally think about this problem. If we're in a reduction scheduler and we're using the rfactored reduction tensor view, we don't care about the rfactor domain, we care about the root domain because we're looking to vectorize the reads (input tensor views). Otherwise we do want the rfactor domain. So this is the reverse of our typical check, we actually want to selectively ignore the rfactor domain.

Reply:
Ah, OK, that makes sense. Can you also leave it as a comment?